internal/slicer/slicer.go

```diff
@@ -218,98 +221,42 @@ func Run(options *RunOptions) (*Report, error) {
 	// Create new content not coming from packages.
 	done := make(map[string]bool)
 	for _, slice := range options.Selection.Slices {
-		arch := archives[slice.Package].Options().Arch
+		archiveName := options.Selection.Release.Packages[slice.Package].Archive
+		arch := options.Archives[archiveName].Options().Arch
 		for relPath, pathInfo := range slice.Contents {
-			if len(pathInfo.Arch) > 0 && !contains(pathInfo.Arch, arch) {
+			if len(pathInfo.Arch) > 0 && !slices.Contains(pathInfo.Arch, arch) {
 				continue
 			}
 			if done[relPath] || pathInfo.Kind == setup.CopyPath || pathInfo.Kind == setup.GlobPath {
 				continue
 			}
 			done[relPath] = true
-			addKnownPath(relPath)
+			addParents(knownPaths, relPath)
```

Comment:
I've been looking at this PR for a while, but I have to say I'm quite uncomfortable with the diff as is. It's not that it is wrong, but rather that it is difficult to understand if it is wrong or not.  Major pieces are being moved around, which was the plan, but also once again we have semantic changes that were not supposed to be bundled with a major move, as it makes it reviewing hard to impossible. For example, here we used to add a known path and all its parents, but now we're simply adding parents. Why?  Elsewhere we have a reviewer note claiming we now do not trust parents to have been added because maybe they weren't, but that was an impossible situation on the current design because we always add known paths together with their own parents, thus every time you hit a parent you know their preceding parents have also been added.

Can we please take a step back and do as we discussed before: major code reorganizations should be done independently from semantic changes, otherwise we can easily introduce hard to track bugs as has happened recently elsewhere.

Then, there are minor-looking changes that are actually bundled here for no good reason.  For example,  it modifies where the architecture setting is looked at. Why?  How's that related to splitting slicer steps?

We need to take a step back here.  If this is about moving major parts of some complex logic around, let's please not touch anything else while doing this, and let's try to preserve what is being done as much as possible, so our brains can process this in "same thing but elsewhere" mode.  If you want to change other details later, that's fine of course, but then we'll have a chance of reviewing the changes as actual changes.

Reply:
Per the discussion we had today I will remove all the changes to the logic, namely:
* Combine adding the parent paths to known paths and the path data into one function https://github.com/canonical/chisel/pull/136/commits/f4dbed46ebce2378d034f82edd5aec605b0c21b0. 
* Remove the `fetchArchives` function and do it in a separate PR later https://github.com/canonical/chisel/pull/136/commits/25d52ac6178136ea27adb5b633c266d06a81b06c. 
* Bring back the `archives` map to get the `arch` https://github.com/canonical/chisel/pull/136/commits/6ea272c48a6a63ba1dcc734ba3548e31504105f0.